bc_obps/registration/schema/v1/facility.py

```diff
@@ -0,0 +1,114 @@
+                    "municipality": values.municipality,
+                    "province": values.province,
+                    "postal_code": values.postal_code,
+                }
+            )
+        return data
+
+    class Config:
+        model = Facility
+        model_fields = [
+            "address",
+            "latitude_of_largest_emissions",
+            "longitude_of_largest_emissions",
+        ]
+
+
+class FacilityIn(ModelSchema):
+    street_address: Optional[str] = None
+    municipality: Optional[str] = None
+    province: Optional[str] = None
+    postal_code: Optional[str] = None
+    operation_id: UUID
+    well_authorization_numbers: Optional[list] = None
+
+    @model_validator(mode='after')  # type: ignore
+    def create_address_data(cls, values):
+        payload = {}
+        if hasattr(values, 'street_address') and values.street_address:
+            address_data = values.dict(include={'street_address', 'municipality', 'province', 'postal_code'})
+            payload['address_data'] = address_data
+
+        if hasattr(values, 'well_authorization_numbers') and values.well_authorization_numbers:
+            payload['well_data'] = values.dict(include={'well_authorization_numbers'})['well_authorization_numbers']
+
+        payload['facility_data'] = values.dict(
+            exclude={
+                'street_address',
+                'municipality',
+                'province',
+                'postal_code',
+                'operation_id',
+                'well_authorization_numbers',
+            }
+        )
+
+        payload['operation_id'] = values.dict(include={'operation_id'})['operation_id']
+
+        return payload
+
+    class Config:
+        model = Facility
+        model_fields = [
+            "name",
+            "type",
+            "latitude_of_largest_emissions",
+            "longitude_of_largest_emissions",
+            "well_authorization_numbers",
+        ]
+        populate_by_name = True
+
+
+class FacilityOut(Schema):
+    section1: FacilitySection1
+    section2: FacilitySection2
+    id: UUID
+
+    @staticmethod
+    def resolve_section1(obj: Facility) -> Dict[str, Any]:
+        return {"name": obj.name, "type": obj.type, 'well_authorization_numbers': obj.well_authorization_numbers}
+
+    @staticmethod
+    def resolve_section2(obj: Facility) -> Dict[str, Union[Decimal, str, None]]:
+        data: Dict[str, Union[Decimal, str, None]] = {
+            "latitude_of_largest_emissions": obj.latitude_of_largest_emissions,
+            "longitude_of_largest_emissions": obj.longitude_of_largest_emissions,
+        }
+        if obj.address:
+            data.update(
+                {
+                    "street_address": obj.address.street_address,
+                    "municipality": obj.address.municipality,
+                    "province": obj.address.province,
+                    "postal_code": obj.address.postal_code,
+                }
+            )
+        return data
```

Comment:
We wouldn't need all this if we didn't use sections in the `jsonSchema`

Reply:
As we discussed that with Suha yesterday it sounds like the green checks will stay in though if we run into performance issues we will validate on the existing formData on mount and also on submit.  
  
I think reporting will have to use nested data as well and not just for this validation check mark so I appreciate you being the first to sort this out 🥇 It should be a lot easier to do going forward!